src/Microsoft.AspNet.Mvc.Core/ResponseCacheAttribute.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Globalization;
+
+namespace Microsoft.AspNet.Mvc
+{
+    /// <summary>
+    /// An <see cref="ActionFilterAttribute"/> which sets the appropriate headers related to Response caching.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Class | AttributeTargets.Method, AllowMultiple = false, Inherited = true)]
+    public class ResponseCacheAttribute : ActionFilterAttribute
+    {
+        // A nullable-int cannot be used as an Attribute parameter.
+        // Hence this nullable-int is present to back the Duration property.
+        private int? _duration;
+        
+        /// <summary>
+        /// Gets or sets the duration in seconds for which the response is cached.
+        /// </summary>
+        public int Duration
```

Comment:
One problem with this approach is that you cannot un-set the value. Once set, the user can never cause `_duration == null` to occur, and so a `max-age` will always be written. I don't think it's a big problem, because explicit `max-age=0` is pretty common.